ios/library/WhirlyGlobe-MaplyComponent/src/control/MaplyViewController.mm

```diff
@@ -354,31 +355,44 @@ - (void) loadSetup_lighting
 
 - (ViewRef) loadSetup_view
 {
-    if (_coordSys)
-    {
-        const auto bbox = [_coordSys getBounds];
-        const auto ll3d = Point3d(bbox.ll.x, bbox.ll.y, 0);
-        const auto ur3d = Point3d(bbox.ur.x, bbox.ur.y, 0);
-        const Point3d diff = ur3d - ll3d;
-        const auto center3d = Point3d(_displayCenter.x,_displayCenter.y,_displayCenter.z);
-        // May need to scale this to the space we're expecting
-        const auto scaleFactor = (std::abs(diff.x()) > 10.0 || std::abs(diff.y()) > 10.0) ?
-                                    4.0/std::max(diff.x(),diff.y()) : 1.0;
-        coordAdapter = std::make_shared<GeneralCoordSystemDisplayAdapter>(
-              [_coordSys getCoordSystem].get(),ll3d,ur3d,center3d,
-              Point3d(scaleFactor,scaleFactor,1));
-    } else {
+    if (mapType == MaplyMapTypeOverlay) {
+        // In this case the view is tied to an outside matrix
         const auto originLon = 0.0;
         const auto ll = GeoCoord::CoordFromDegrees(-180.0,-90.0);
         const auto ur = GeoCoord::CoordFromDegrees(180.0,90.0);
         coordAdapter = std::make_shared<SphericalMercatorDisplayAdapter>(originLon, ll, ur);
+
+        mapView = std::make_shared<MapViewOverlay_iOS>(coordAdapter.get());
+        mapView->setContinuousZoom(true);
+        mapView->setWrap(_viewWrap);
+        mapView->addWatcher(animWrapper);
+    } else {
+        if (_coordSys)
+        {
+            const auto bbox = [_coordSys getBounds];
+            const auto ll3d = Point3d(bbox.ll.x, bbox.ll.y, 0);
+            const auto ur3d = Point3d(bbox.ur.x, bbox.ur.y, 0);
+            const Point3d diff = ur3d - ll3d;
+            const auto center3d = Point3d(_displayCenter.x,_displayCenter.y,_displayCenter.z);
+            // May need to scale this to the space we're expecting
+            const auto scaleFactor = (std::abs(diff.x()) > 10.0 || std::abs(diff.y()) > 10.0) ?
+            4.0/std::max(diff.x(),diff.y()) : 1.0;
+            coordAdapter = std::make_shared<GeneralCoordSystemDisplayAdapter>(
+                                                                              [_coordSys getCoordSystem].get(),ll3d,ur3d,center3d,
+                                                                              Point3d(scaleFactor,scaleFactor,1));
+        } else {
+            const auto originLon = 0.0;
```

Comment:
Looks like this duplicate block could be eliminated with `|| !_coordSys` on the outer `if`.